examples/parallel_read.py

```diff
@@ -0,0 +1,74 @@
+# This file is to test collective io in h5py
+ 
+"""
+Author:  Jialin Liu, jalnliu@lbl.gov
+Date:    July 11, 2017
+Prerequisites: python 2.7.0, mpi4py and numpy
+This code does parallel read (a 2D array) with independent I/O or collective IO
+Parameters: filename datasetname is_collective
+Example: mpirun -np 10 python-mpi parallel_read.py test.h5 dset 0 
+0 for independent IO
+1 for collective IO
+"""
+
+from mpi4py import MPI
+import numpy as np
+import h5py
+import time
+import sys
+
+if len(sys.argv)>3:
+   filename=str(sys.argv[1])
+   dataset =str(sys.argv[2])
+   colr = int(sys.argv[3])
+else:
+   print "args: filename dataset is_collective?"
+   sys.exit()
+comm =MPI.COMM_WORLD
+nproc = comm.Get_size()
+f = h5py.File(filename, 'r', driver='mpio', comm=MPI.COMM_WORLD)
+rank = comm.Get_rank()
+#open the dataset, get the handle, shape
+dset = f[dataset]
+
+
+length_x = dset.shape[0]
+length_y = dset.shape[1]
+if rank==0:
+   print (length_x,length_y)
+
+f.atomic = False # for better performance
+
+#devide the workload, 
```

Comment:
You want something like `git rebase --onto h5py/master start-branch end-branch`, where `h5py` is the name of the remote which points to this repo, `start-branch` is the point where you diverged from what's in this repos master, and `end-branch` is your current branch. If you can't get this working, I can clean this up, but if you can do the rest that'd be helpful.